llvm/lib/Target/RISCV/RISCVFeatures.td

```diff
@@ -599,15 +599,21 @@ def HasStdExtZvkned : Predicate<"Subtarget->hasStdExtZvkned()">,
 def FeatureStdExtZvknha
     : SubtargetFeature<"experimental-zvknha", "HasStdExtZvknha", "true",
                        "'Zvknha' (Vector SHA-2 (SHA-256 only))">;
-
-def FeatureStdExtZvknhb
-    : SubtargetFeature<"experimental-zvknhb", "HasStdExtZvknhb", "true",
-                       "'Zvknhb' (Vector SHA-2 (SHA-256 and SHA-512))",
-                       [FeatureStdExtZvknha]>;
 def HasStdExtZvknha : Predicate<"Subtarget->hasStdExtZvknha()">,
                                 AssemblerPredicate<(all_of FeatureStdExtZvknha),
                                 "'Zvknha' (Vector SHA-2 (SHA-256 only))">;
 
+def FeatureStdExtZvknhb
+    : SubtargetFeature<"experimental-zvknhb", "HasStdExtZvknhb", "true",
+                       "'Zvknhb' (Vector SHA-2 (SHA-256 and SHA-512))">;
```

Comment:
I think that Zvknhb requires Zve64x

Reply:
`Zvknhb` requires `Zve64x` but doesn't **imply** `Zvk64x lol~

Reply:
I think "requires" might mean "implies". This keeps happening with different extensions. Here's one example https://github.com/riscv/riscv-v-spec/issues/723#issuecomment-922153867

Reply:
If so, why should the check like this exist?
https://github.com/llvm/llvm-project/blob/69074bf144c1bf54040d289584e3d1d54f883a37/llvm/lib/Support/RISCVISAInfo.cpp#L935

Reply:
> If so, why should the check like this exist?
> 
> https://github.com/llvm/llvm-project/blob/69074bf144c1bf54040d289584e3d1d54f883a37/llvm/lib/Support/RISCVISAInfo.cpp#L935

That's a good question. That one is weird because Zve32x implies Zvl32b, but Zvl32b requires at least Zve32x. So they're mutually dependent.

Reply:
And I should also remove these [checks](https://github.com/llvm/llvm-project/blob/69074bf144c1bf54040d289584e3d1d54f883a37/llvm/lib/Support/RISCVISAInfo.cpp#L940C1-L940C1) and add them to **ImpliedExts***?

Reply:
maybe we should ask for clarification in https://github.com/riscv/riscv-crypto

Reply:
Sure, maybe I can ask Nicolas directly, but maybe it should be handle in another patch.

Reply:
My colleague just show me the [spec](https://github.com/riscv/riscv-crypto/blob/main/doc/vector/riscv-crypto-spec-vector.adoc#1-extensions-overview) has mentioned: "The [Zvknhb](https://github.com/llvm/llvm-project/pull/66860#zvknh) and [[zvbc]](https://github.com/llvm/llvm-project/pull/66860#zvbc) Vector Crypto Extensions --and accordingly the composite extensions [[Zvkn]](https://github.com/llvm/llvm-project/pull/66860#Zvkn) and [[Zvks]](https://github.com/llvm/llvm-project/pull/66860#Zvks)-- require a Zve64x base, or application ("V") base Vector Extension."
so I think those checks should be moved to **ImpliedExts***.